File: src/main/java/com/jss/jssbackend/modules/miscellaneous/service/CountryService.java
package com.jss.jssbackend.modules.miscellaneous.service;

import java.util.List;

import com.jss.jssbackend.modules.miscellaneous.model.Country;

public interface CountryService {
    public List<Country> getCountries();

    public Country getCountry(Integer id);
}

File: client-osiris/src/app/modules/profile/services/country.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../main/services/appRest.service';
import { Country } from '../../profile/model/Country';

@Injectable({
  providedIn: 'root'
})
export class CountryService extends AppRestService<Country> {

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getCountries() {
    return this.getList(new HttpParams(), "countries");
  }
}
